src/trix/controllers/composition_controller.coffee

```diff
@@ -42,11 +42,13 @@ class Trix.CompositionController extends Trix.BasicObject
       @documentView.render()
       @revision = @composition.revision
 
-    unless @documentView.isSynced()
-      @delegate?.compositionControllerWillSyncDocumentView?()
-      @documentView.sync()
-      @reinstallAttachmentEditor()
-      @delegate?.compositionControllerDidSyncDocumentView?()
+    @element.shadowing = @attachmentEditor?
```

Comment:
Using this property to indicate that the shadow element should be used for serializing. Setting it here is less than ideal, but it works for testing the idea.